---
title: "Multi Omic Data explioration"
author: "Edmond Géraud Aguilar"
date: '`r format(Sys.Date(),"%e de %B, %Y")`'
output:
 prettydoc::html_pretty:
    toc: true
    theme: cayman
    highlight: github
    number_sections: true
bibliography: references.bibtex
link-citations: true
linkcolor: blue
editor_options: 
  chunk_output_type: inline
---


```{r setup, include=FALSE}
# knitr options
knitr::opts_chunk$set(comment = NA, prompt = TRUE, tidy.opts=list(width.cutoff=60),tidy=TRUE, 
               fig.width = 7, fig.height = 7,echo = TRUE, 
               message = T, warning =T, cache=T,fig.pos = "!H", out.extra = "")
```

```{r librerias}
library(GEOquery)
library(ggplot2)
library(limma)
library(DESeq2)
library(Rsubread)
library(data.table)
library(edgeR)
library(ggrepel)
library(sva)
```

# Description of original publication of data-set to be applied


## Setup environment

```{r setupNevironment}
dir.create("./results")
```



## Metadata analysis


```{r downloadMetaData}
metadata.file<-"./data/rnaseqData/metadata.csv"
metadata.raw<-read.csv(file = metadata.file,header = T,sep = ",",dec = ".")
w<-unlist(apply(metadata.raw, 2, function(x) which(is.na(x))))
metadata<-metadata.raw[-w,]
# kableExtra::kable(metadata)
kableExtra::kable(metadata)

```

### Descriptive analysis  

We have gender, age at death, post mortem hours, Braak and Cerad soroes, integrity numbers for RNA, and percentage of neurons (they don't specify what it is)

Let's see first classify cause of death by complications of AD and others, and also, convert into factors, group, gender, Cerad, Braak into factors.

```{r preparandoMetadata}
colnames(metadata)<-c("ID","Group","Gender","AgeDeath","PMI","Braak","Cerad","RIN_RNA","neuron","CauseDeath") ## column names chaged for easy manipulation

## Classifiying cause of death
w<-grep("AD",metadata$CauseDeath)
metadata$CauseDeath[w]<-"ADcomp"
metadata$CauseDeath[-w]<-"Other"
## characters to factos
for(j in 1:ncol(metadata)){
  
  if(class(metadata[,j])=="character"){
    metadata[,j]<-as.factor(metadata[,j])
  }
  
}
```

Let's se how the data is finally
```{r preparandoMetadata2}
metadata.copy<-metadata[,-1]# let's remove the index
str(metadata.copy)
```


First of all let's observe the min, max, the quartiles, and the mean, for numerical variables, and the categorical variables, his levels for each group.

```{r}
sumarizacion<-function(datos,grupo){
  
  datos.grupo<-datos[datos$Group==grupo,]
  datos.grupo$Group<-droplevels(datos.grupo$Group)
  s<-summary(datos.grupo)
  s[is.na(s)]<-""
  lista<-list("summary"=s,"data"=datos.grupo)
  return(lista)
  
  
}
young<-sumarizacion(metadata.copy,"Young")
kableExtra::kable(young$summary)
```
```{r}
par(mfrow=c(2,2))
for(i in 1:ncol(young$data)){
  
  if(class(young$data[,i])!="factor"){
    
    hist(young$data[,i],main=colnames(young$data)[i],freq=FALSE,xlab="Young n=8")
    lines(density(young$data[,i]),col="blue")
    
  }
  
}
```

```{r}
par(mfrow=c(2,2))
for(i in 1:ncol(young$data)){
  
  if(class(young$data[,i])!="factor"){
    
    boxplot(young$data[,i],main=colnames(young$data)[i],freq=FALSE,xlab="Young n=8")
    points(mean(young$data[,i]),pch=7)

  }
  
}
```



```{r}
old<-sumarizacion(metadata.copy,"Old")
kableExtra::kable(old$summary)
```

```{r}
par(mfrow=c(2,2))
for(i in 1:ncol(old$data)){
  
  if(class(old$data[,i])!="factor"){
    
    hist(old$data[,i],main=colnames(old$data)[i],freq=FALSE,xlab="old n=10")
    lines(density(old$data[,i]),col="blue")
    
  }
  
}
```
```{r}
par(mfrow=c(2,2))
for(i in 1:ncol(old$data)){
  
  if(class(old$data[,i])!="factor"){
    
    boxplot(old$data[,i],main=colnames(old$data)[i],freq=FALSE,xlab="old n=10")
    points(mean(old$data[,i]),pch=7)

  }
  
}
```


```{r}
ad<-sumarizacion(metadata.copy,"AD")
kableExtra::kable(ad$summary)
```

```{r}
par(mfrow=c(2,2))
for(i in 1:ncol(ad$data)){
  
  if(class(ad$data[,i])!="factor"){
    
    hist(ad$data[,i],main=colnames(ad$data)[i],freq=FALSE,xlab="ad n=12")
    lines(density(ad$data[,i]),col="blue")
    
  }
  
}
```

```{r}
par(mfrow=c(2,2))
for(i in 1:ncol(ad$data)){
  
  if(class(ad$data[,i])!="factor"){
    
    boxplot(ad$data[,i],main=colnames(ad$data)[i],freq=FALSE,xlab="ad n=12")
    points(mean(ad$data[,i]),pch=7)

  }
  
}
```


At first sight one can observe that the PMI feature, is some how imbalance, in the way that the mean and the median are not close to each others. For the other numerical variables, it seems that the data follows a normal distribution, given the fact that the mean and the median are close to each other. We have 12 patients with AD, and 18 controls, 10 Old and 8 Young. We have 3 females and 27 males. From the Braak stage, we have I/II in 5 samples. and 12 samples in V/VI stage. From Cerad score, we only have the 12 patients with AD. 



```{r plotsContVarvsGroups}
par(mfrow=c(2,2))
for ( i in 1:dim(metadata.copy)[2]) {
  if (class(metadata.copy[,i])!="factor"){
      boxplot(metadata.copy[,i]~metadata.copy$Group,main=colnames(metadata.copy)[i],
              ylab = colnames(metadata.copy)[i],xlab = "")
      means<-tapply(metadata.copy[,i], metadata.copy$Group, mean)
      points(means,col="blue")
    
    }
}

```


```{r plotsContrlCategorical}

x<-metadata.copy[,c(2,5,6,9)]
y<-apply(x, 2, function(column) table(metadata.copy$Group,column))
par(mfrow=c(2,2))
for( i in 1:length(y)){
  y.t<-y[[i]]
  barplot(y.t,xlim=c(0,ncol(y.t)+3),
          col=1:3,
          ylab="Subjects",
          legend.text=T,
          args.legend=list(
            x=ncol(y.t)+3,
            y=max(colSums(y.t)),
            bty="n"
          ))
  
  
}
```


#### (EXTRA IF NEEDED) Analysis betwwen groups and other features

```{r testhomogeneityNormality}
with(metadata.copy,bartlett.test(AgeDeath~Group))

with(metadata.copy,bartlett.test(PMI~Group))

with(metadata.copy,bartlett.test(RIN_RNA~Group))

with(metadata.copy,bartlett.test(neuron~Group))

with(metadata.copy,shapiro.test(AgeDeath))

with(metadata.copy,shapiro.test(PMI))

with(metadata.copy,shapiro.test(RIN_RNA))

with(metadata.copy,shapiro.test(neuron))


```
It seems that there is no heterogeneity of variances, and although there are only 30 samples, Shapiro-Wilks test tell us that the normality holds. Thus we are going to perform anova test with Bonferroni correction in all continuous features compared with groups.

```{r aovDeath}
summary(with(metadata.copy,aov(AgeDeath~Group)))
pairwise.t.test(metadata.copy$AgeDeath,metadata.copy$Group,method="bonferroni")
```
Performing anova over the controls and AD subjects, we can observe that there are statistican significance between old , young, AD, young subjects. But there is not statistical significance between Old and AD subjects.
```{r anovaPMI}
summary(with(metadata.copy,aov(PMI~Group)))
```
Anova indicating that there is not significance evidence between the post mortem interval between subjects. 
```{r rinRNAanova}
summary(with(metadata.copy,aov(RIN_RNA~Group)))
pairwise.t.test(metadata.copy$RIN_RNA,metadata.copy$Group,method="bonferroni")
```
The integrity numbers of RNA shows a statsitical evidence between AD and Young subjects. But there is no difference between old and young controls. 
```{r statisticalAnalysisMetadata}
summary(with(metadata.copy,aov(neuron~Group)))
pairwise.t.test(metadata.copy$neuron,metadata.copy$Group,method="bonferroni")
```



# RNA-seq exploratory analysis


## Loading the superseries and RNA-seq subseries
```{r cargaDtos}
# gse.superserie<-getGEO("GSE153875",destdir = "data/superserie",GSEMatrix = T)
# gse.GSE130746<-getGEO("GSE130746",destdir = "ADdata/subseries/gse130746_chip_seq/",GSEMatrix = T) #chip-seq
gse.GSE153873<-getGEO("GSE153873",destdir = "data/rnaseqData/",GSEMatrix = T) # RNA-seq
counts<-read.csv("./data/rnaseqData/GSE153873_summary_count.star.txt",header = T,sep = "\t",row.names="refGene")
```
### INVESTIGATION ABOUT THE UNKOWN FEATURE
First let's investigate which T and A refer, and how they possible lines to metadata. Furthermore lets contrusct this data frame as a "targets" in which will go the pheno data.

```{r targetsFile}
title<-gse.GSE153873$GSE153873_series_matrix.txt.gz$title
title.temp<-matrix(strsplit2(title,"-"),ncol=3)
title.temp.df<-as.data.frame(title.temp)
colnames(title.temp.df)<-c("ID","dontknow","Group")

title.temp.df$shortname<-paste(title.temp.df$Group,title.temp.df$dontknow,title.temp.df$ID,sep="_")
title.temp.df$ID<-as.numeric(title.temp.df$ID)
title.temp.df.ordered<-title.temp.df[order(title.temp.df$ID),]
title.temp.df.ordered$dontknowID<-as.numeric(rownames(title.temp.df.ordered))
w<-which(title.temp.df.ordered$dontknow%in%title.temp.df.ordered$dontknow[grep("A",title.temp.df.ordered$dontknow)])
title.temp.df.ordered$dontknowLetter<-numeric(length(nrow(title.temp)))
title.temp.df.ordered$dontknowLetter[w]<-"A"
title.temp.df.ordered$dontknowLetter[-w]<-"T"
targets.temp<-cbind(title.temp.df.ordered,metadata.copy)
targets.temp2<-targets.temp[,-7]
levels(targets.temp2$Gender)<-c("F","M")
targets.temp2$shortname<-paste(targets.temp2$Group,targets.temp2$dontknow,targets.temp2$Gender,targets.temp2$Braak,targets.temp2$ID,sep="_")
targets<-targets.temp2
```



```{r reordering}
write.csv(targets,file="./results/targets.csv",sep=",")
coldata<-targets
for( j in 1:ncol(targets)){
  
  if(class(targets[,j])=="character"){
    coldata[,j]<-as.factor(coldata[,j])
  }
}

col.namez<-strsplit2(colnames(counts),split="[.]")
col.namez[,1]<-gsub("X","",col.namez[,1])
col.namez.df<-as.data.frame(col.namez)
colnames(col.namez.df)<-c("ID","dontknow","Group")
col.namez.df$ID<-as.numeric(col.namez.df$ID)
counts.order<-counts[,order(col.namez.df$ID)]
col.namez.df$shortName<-paste(col.namez.df$Group,col.namez.df$dontknow,col.namez.df$ID,sep="_")
col.namez.df$shortName.ordered<-col.namez.df$shortName[order(col.namez.df$ID)]
counts.order<-counts[,order(col.namez.df$ID)]
colnames(counts.order)<-col.namez.df$shortName.ordered
colnames(counts.order)<-targets$shortname
```



```{r investigandoMisterioso}
par(mfrow=c(2,2))
for ( i in 1:dim(targets)[2]) {
  if (class(targets[,i])=="factor"){
    if(colnames(targets)[i]!="dontknow"){
      boxplot(targets$dontknowID~targets[,i],main=colnames(targets)[i],
              ylab = "Dontknow ID",xlab = "")
      # means<-tapply(targets[,i], targets[,i], mean)
      # points(means,col="blue")
    
    }
}}


```


```{r investigandoMAsvariableMisteriosaN}
grupo<-table(targets$dontknowLetter,targets$Group)
genero<-table(targets$dontknowLetter,targets$Gender)
braak<-table(targets$dontknowLetter,targets$Braak)
cerad<-table(targets$dontknowLetter,targets$Cerad)
y<-list("grupo"=grupo,"genero"=genero,"braak"=braak,"cerad"=cerad)
par(mfrow=c(2,2))
for( i in 1:length(y)){
  y.t<-y[[i]]
  barplot(y.t,xlim=c(0,ncol(y.t)+3),
          col=1:3,
          ylab="DontKnowLeter",
          legend.text=T,
          args.legend=list(
            x=ncol(y.t)+3,
            y=max(colSums(y.t)),
            bty="n"
          ))
  
  
}
```
## RNA-seq data exploration

```{r}
head(counts.order)
```


As a rule of thumb, genes are dropped if they can’t possibly be expressed in all the samples for any of the conditions. Users can set their own definition of genes being expressed. Usually a gene is required to have a count of 5-10 in a library to be considered expressed in that library. Users should also filter with count-per-million (CPM) rather than filtering on the counts directly, as the latter does not account for differences in library sizes between samples

First we create an structured list.
```{r}
dge <- DGEList(counts=counts.order,group = targets$Group,genes = rownames(counts.order))
```

```{r}
dge$samples
```

### Filter

We filter out lowly expressed genes:

The `filterByExpr` function keeps rows that have worthwhile counts in a minimum of samples, 8 samples in our case. The function access the `group` factor contained in `dge` in order to compute the miimum group size, but the filtering is performed independently of which sample belongs to which group so that no bias is introduced. It is recommended to recalculate the library sizes of the `DGEList` object
after the filtering, although the downstream analysis is robust to whether this is done or not.


```{r}
keep <- filterByExpr(dge)
dge_filt<-dge[keep, ,keep.lib.sizes=F]
dge_filt$samples
```
### Normalization TMM

```{r}
dge_norm<-calcNormFactors(dge_filt)
```

```{r}
ratio<-max(dge_norm$samples$lib.size)/min(dge_norm$samples$lib.size)
ratio
```

```{r}

v<-voom(dge_norm,plot=T)
```
```{r}
boxplot(v$E,cex.axis = 0.5, las = 2, which = "all", col = c(rep("red", 
  8), rep("blue", 10), rep("green", 12)), main = "Distribution of log counts",names=rownames(v$targets))
```

### PCA

```{r}
plotPCA3 <- function (datos, labels, factor, title, scale,center,colores, size = 1.5, glineas = 0.25) {
  data <- prcomp(t(datos),scale=scale,center=center)
  # plot adjustments
  dataDf <- data.frame(data$x)
  Group <- factor
  loads <- round(data$sdev^2/sum(data$sdev^2)*100,1)
  # main plot
  p1 <- ggplot(dataDf,aes(x=PC1, y=PC2)) +
    theme_classic() +
    geom_hline(yintercept = 0, color = "gray70") +
    geom_vline(xintercept = 0, color = "gray70") +
    geom_point(aes(color = Group), alpha = 0.55, size = 3) +
    coord_cartesian(xlim = c(min(data$x[,1])-5,max(data$x[,1])+5)) +
    scale_fill_discrete(name = "Group")
  # avoiding labels superposition
 grafico<- p1 + geom_text_repel(aes(y = PC2 + 0.25, label = labels),segment.size = 0.25, size = size) + 
    labs(x = c(paste("PC1",loads[1],"%")),y=c(paste("PC2",loads[2],"%"))) +  
    ggtitle(paste("Principal Component Analysis for: ",title,sep=" "))+ 
    theme(plot.title = element_text(hjust = 0.5)) +
    scale_color_manual(values=colores)
    return(list("loads"=loads,"grafico"=grafico))
  }
```


```{r}
loads<-plotPCA3(v$E,labels=rownames(v$targets),factor = v$targets$group,title="PCA normalized data",scale = F,center = T,size=3,colores=1:7)
loads$grafico
```



```{r}
adj_counts<-ComBat_seq(dge_norm$counts,batch = targets$dontknowLetter,group=targets$Group)
```




```{r}
dge_norm.copy<-dge_norm
dge_norm.copy$counts<-adj_counts
myvoom<-voom(dge_norm.copy,plot=T)
```


```{r}
boxplot(myvoom$E,cex.axis = 0.5, las = 2, which = "all", col = c(rep("red", 
  8), rep("blue", 10), rep("green", 12)), main = "Distribution of log counts",names=rownames(myvoom$targets))

```



```{r}
loads<-plotPCA3(myvoom$E,labels=rownames(myvoom$targets),factor = myvoom$targets$group,title="PCA normalized data",scale = F,center = T,size=3,colores=1:7)
loads$grafico
```

